### PR TITLE
Skip checking codespace and abci code if running compatibility tests.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Adds an `IsCompatibilityTest` to allow checking if we're running one. Added as a function for potential re-use elsewhere.

closes: #3653

### Commit Message / Changelog Entry

```bash
chore: skip checking the codespace and abci code of the expected error in AssertTxSuccess if running compatibility tests
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/testconfig/testconfig.go [hunk 1]
```diff
@@ -323,6 +323,13 @@ func GetChainBTag() string {
 	return GetChainATag()
 }
 
+// IsCompatibilityTest returns true if the test is running against two different
+// chain versions, false is returned if the test is running against the same
+// chain version.
+func IsCompatibilityTest() bool {
+	return GetChainATag() != GetChainBTag()
```
[the remark this answers is not in the document]
Comment (DimitrisJim) [reply e2e/testconfig/testconfig.go:330]: I think the naming could be tweaked? The tags being should correctly signify a compat test but tags being the same don't always mean we aren't in one. (Looking at the `.json` files we test against ourselves).

[the remark this answers is not in the document]
Comment (crodriguezvega) [reply e2e/testconfig/testconfig.go:330]: Yes, so what will happen if we are running a compatibility test but tags are the same?

[the remark this answers is not in the document]
Comment (DimitrisJim) [reply e2e/testconfig/testconfig.go:330]: In that case we'd basically have no compatibility issues. I guess this doesn't really signify if the compatibility _workflow_ is running as much as it signifies if the tags are different.

I.e, is a compatibility test using the same tags a compatibility test?

[the remark this answers is not in the document]
Comment (chatton) [reply e2e/testconfig/testconfig.go:330]: I guess we don't actually need this to happen in all compatibility tests, we just need it to happen if `GetChainATag() != GetChainBTag()`. Maybe we can just do that check directly in the `AssertTxFailure` fn where it's needed?

**Review by colin-axner — APPROVED** [review]
Thanks!

**Review by chatton — APPROVED** [review]
LGTM maybe we can create an issue for @colin-axner 's suggestion

**Review by damiannolan — APPROVED** [review]

**Comment by colin-axner** [comment]
> LGTM maybe we can create an issue for @colin-axner 's suggestion

Added to #3663 https://github.com/cosmos/ibc-go/issues/3663#issuecomment-1572248702 as I think that's when it would be useful

**Pull request merged**